pkg/client/vault/client.go

```diff
@@ -102,23 +136,73 @@ func (c *Client) CreateVault() (*CreatedVault, error) {
 	}
 
 	return &CreatedVault{
-		ID:  didKey,
-		KMS: kmsLoc,
-		EDV: edvLoc,
+		ID: didKey,
+		Authorization: Authorization{
+			KMS: &Location{
+				URI:  kmsURI,
+				ZCAP: kmsZCAP,
+			},
+			EDV: edvLoc,
+		},
 	}, nil
 }
 
-// CreatedVault represents success response of CreateVault function.
-type CreatedVault struct {
-	ID  string    `json:"id"`
-	EDV *Location `json:"edv"`
-	KMS *Location `json:"kms"`
+// SaveDoc saves a document by encrypting it and storing it in the vault.
+func (c *Client) SaveDoc(vaultID, id string, content interface{}, auth *Authorization) (*SavedDoc, error) {
+	key, encContent, err := encryptContent(content)
+	if err != nil {
+		return nil, fmt.Errorf("encrypt content: %w", err)
+	}
+
+	encKey, err := encryptKey(c.webKMS(vaultID, auth.KMS), c.webCrypto(vaultID, auth.KMS), key)
+	if err != nil {
+		return nil, fmt.Errorf("encrypt key: %w", err)
+	}
+
+	payload, err := sjson.Set(encKey, "content", base64.StdEncoding.EncodeToString(encContent))
```

Comment:
you don't need to go through the hassle of building the encrypted content and a shared secret separately here.

Use AFG directly to build the JWE envelope that contains the encrypted content and the wrapped cek all in one envelope.

I don't get why the encrypted key (wrapped cek)  need to be used as an index in `payload`.

cc @troyronda @DRK3

Reply:
@Baha-sk  you mean like I did here https://github.com/trustbloc/edge-service/pull/584/files#diff-286763af51a9a452468f5d071c9f2323ddac0dba0ce776c7ddc73ef9bad3662aR329